verification: `cargo verify` doesn't propagate `cargo build` error codes

`cargo verify` on current main (https://github.com/coconut-svsm/svsm/commit/697e6c5b10665991dca4ff46c8714c2facea835c) fails but it returns 0, so the CI is not failing.

It seems that `cargo verify` doesn't propagate `cargo build` error status:
```
$ cargo verify; echo "\ncargo verify - exit code: $?" 
RUSTC="verus-rustc" RUSTUP_TOOLCHAIN="1.91.0" VERUS="/home/stefano/.cargo/bin/verus" VERUS_TARGETS="svsm,verify_external,verify_proof,verus_stub,vstd" "cargo" "build" "--features" "verus"
   Compiling svsm v0.1.0 (/home/stefano/repos/coconut/svsm/kernel)
error[E0425]: cannot find value `item_size` in this scope
    --> kernel/src/mm/alloc.rs:1034:28
     |
1034 |                 item_size: item_size as u64,
     |                            ^^^^^^^^^ not found in this scope

For more information about this error, try `rustc --explain E0425`.
error: could not compile `svsm` (lib) due to 1 previous error
cargo build failed with status: exit status: 101

cargo verify - exit code: 0
```

@ziqiaozhou @Chris-Hawblitzel I'm not sure where to open the issue, so for now I'm opening it here.

--- a/.github/workflows/manual-verify.yml
+++ b/.github/workflows/manual-verify.yml
@@ -24,10 +24,18 @@ jobs:
         run: |
           verusfmt --check `find ./ -name *.verus.rs` --verus-only
 
+      # cargo-v doesn't propagate the `cargo build` exit code, so we need
+      # to check stderr for errors.
+      # When https://github.com/microsoft/verismo/issues/23 will be fixed
+      # we can remove it.
       - name: Verify svsm with verus
-        run: cargo verify
+        run: |
+          cargo verify 2>&1 | tee verify.log
+          ! grep -q 'cargo build failed' verify.log
         working-directory: kernel
 
       - name: Verify extra proof libs with verus
-        run: cargo verify
+        run: |
+          cargo verify 2>&1 | tee verify.log
+          ! grep -q 'cargo build failed' verify.log
         working-directory: verification/verify_proof

